### PR TITLE
Fix/74 warnings

### PR DESCRIPTION
**Pull request by dd32**
This is an attempt at fixing #74, this is untested and won't be worked on from here. Please take this PR and run with it.

### CODE REVIEW

**Review by ashfame — APPROVED** [review]

**Review by ashfame — APPROVED** [review]
@psrpinto I will wait for you to take a look & approve it before merging.

**Review by psrpinto — APPROVED** [review]
👌

**Pull request merged**
